### PR TITLE
Clean AI comps on death

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Clean salvage AI on death from none needed comps

Remove:
insulated
Reflect
MovementIgnoreGravity
Tool
Prying
CanMoveInAir
Tag
Access

Add:
Uncloneable
Unrevivable

## Why / Balance
Make it impossible to kill mobs, revive them and take them to frontier.

## How to test
Spawn xeno, kill xeno, watch it loss its comps

## Media
N/A

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
N/A

**Changelog**
:cl:
- tweak: Most mobs can only be stripped of items after they have been killed.

### CODE REVIEW

**Comment by dvir001** [comment]
Need to add:
  - type: Timer
  - type: TimedSpawner

File: Content.Server/_NF/Salvage/SalvageMobRestrictionsSystem.cs [hunk 1]
```diff
@@ -67,11 +80,25 @@ private void OnRemoveGrid(EntityUid uid, SalvageMobRestrictionsGridComponent com
...
+            RemComp<ToolComponent>(uid);
+            RemComp<PryingComponent>(uid);
+            RemComp<CanMoveInAirComponent>(uid);
+            RemComp<TagComponent>(uid);
+            RemComp<AccessComponent>(uid);
+            EnsureComp<UncloneableComponent>(uid);
+            EnsureComp<UnrevivableComponent>(uid);
+            var damageSpec = new DamageSpecifier(_prototypeManager.Index<DamageGroupPrototype>("Genetic"), 300);
+            _damageableSystem.TryChangeDamage(uid, damageSpec);
         }
```
[the remark this answers is not in the document]
Comment (dvir001) [reply Content.Server/_NF/Salvage/SalvageMobRestrictionsSystem.cs:102]: Silicon will need ``cleanCompsOnMobDeath: false`` if this is the case or some other change.

[the remark this answers is not in the document]
Comment (dvir001) [reply Content.Server/_NF/Salvage/SalvageMobRestrictionsSystem.cs]: Redid it as you asked.

**Review by ErhardSteinhauer — APPROVED** [review]
Spiffing. Also: updated changelog.

**Comment by whatston3** [comment]
Works fine, the strippable change is nice.

Fixed the changelog :^)

**Pull request merged**